objgraph.py

```diff
@@ -857,26 +866,38 @@ def _safe_repr(obj):
         return '(unrepresentable)'
 
 
+def _name_or_repr(value):
+    try:
+        result = value.__name__
+    except AttributeError:
+        result = repr(value)[:40]
+
+    if _isinstance(result, str if sys.version_info[0] == 3 else basestring):
```

Comment:
There's a compat redefinition of `basestring = str` on Python 3 at the top of this file, so this can be simplified to `if _isinstance(result, basestring)`.